web/main.py

```diff
@@ -43,17 +62,32 @@ def handler(dataset=None, filename=None):
         logger.warning(f'Invalid dataset "{dataset}"')
         abort(400)
 
+    bucket_name = f'cpg-{dataset}-{BUCKET_SUFFIX}'
+    bucket = storage_client.bucket(bucket_name)
+
     dataset_project_id = dataset_config['projectId']
     members = read_secret(
         dataset_project_id, f'{dataset}-web-access-members-cache'
     ).split(',')
+
     if email not in members:
-        logger.warning(f'{email} is not a member of the {dataset} access group')
-        abort(403)
+        # Second chance: if there's a '.access' file in the first subdirectory,
+        # check if the email is listed there.
+        split_subdir = filename.split('/', maxsplit=1)
+        if len(split_subdir) == 2 and split_subdir[0]:
+            access_list_filename = f'{split_subdir[0]}/.access'
+            blob = bucket.get_blob(access_list_filename)
+            if blob is None:
+                logger.warning(f'{email} is not a member of the {dataset} access group')
+                abort(403)
+            access_list = blob.download_as_string().decode('utf-8').splitlines()
```

Comment:
Does `download_as_string` download it as binary, you've got a decode here which seems strange

Reply:
Yes, indeed. This has been fixed with newer versions of the library, but unfortunately `hail` (which `cpg-utils` depends on) pins an old version of `google-cloud-storage`...

Reply:
This was a fun one to find out 🤦 the d_as_string downloads bytes, and the d_as_bytes doesn't exist 🎉